deployment/deploy_managed_hsm.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+set -e
+
+echo "There are three steps to deploying the managed HSM"
+echo " - Deploy the HSM in azure"
+echo " - Create 3 keys to encrypt the security domain"
+echo " - Download the security domain and activate the HSM"
+
+output=$(az deployment group create \
```

Comment:
Do we have a cleanup step for this somewhere? I think it would error if you try to create the hsm each run and it already exists...

Reply:
The managed HSM is not deployed per run, since it's non-trivial to setup, it's done once and is long lived. So this is as much about documentation of how to spin up one than anything else.

Reply:
We should probably add a README to the workflows dir to explain things like this for future development and other developers not in this PR.

Reply:
Sounds good, would you like it as part of this review or I can write something up with the final PR for encfs?

Reply:
I think it should be fine to include with the ENCFS PR